src/lib.rs

```diff
@@ -834,6 +876,18 @@ mod env_test {
             "${VAR} ${VAR}" => "value value",
             "${VAR}$VAR" => "valuevalue",
 
+            // default values
+            "/answer/${UNKNOWN:-42}" => "/answer/42",
+            "/answer/${:-42}" => "/answer/${:-42}",
+            "/whatever/${UNKNOWN:-other}$VAR" => "/whatever/othervalue",
+            "/whatever/${UNKNOWN:-other}/$VAR" => "/whatever/other/value",
+            ":-/whatever/${UNKNOWN:-other}/$VAR :-" => ":-/whatever/other/value :-",
+            "/whatever/${VAR:-other}" => "/whatever/value",
+            "/whatever/${VAR:-other}$VAR" => "/whatever/valuevalue",
+            "/whatever/${VAR} :-" => "/whatever/value :-",
+            "/whatever/${:-}" => "/whatever/${:-}",
+            "/whatever/${UNKNOWN:-}" => "/whatever/${UNKNOWN:-}",
```

Comment:
Hm, I'd actually expect this to result in an empty value:
```
~ > bash -c 'echo "Hello ${UNEXPECTED:-} world"'
Hello  world
```

Reply:
fixed